pkg/controller/core/workload_controller.go

```diff
@@ -259,18 +263,55 @@ func (r *WorkloadReconciler) reconcileSyncAdmissionChecks(ctx context.Context, w
 		return false, err
 	}
 
-	queueAdmissionChecks := queue.Spec.AdmissionChecks
-	newChecks, shouldUpdate := syncAdmissionCheckConditions(wl.Status.AdmissionChecks, queueAdmissionChecks)
+	var admissionChecks []string
+	if len(queue.Spec.AdmissionChecks) == 0 {
+		admissionChecks = getChecksFromACStrategies(ctx, wl, &queue)
+	} else {
+		admissionChecks = queue.Spec.AdmissionChecks
+	}
+
+	newChecks, shouldUpdate := syncAdmissionCheckConditions(wl.Status.AdmissionChecks, admissionChecks)
 	if shouldUpdate {
 		log := ctrl.LoggerFrom(ctx)
-		log.V(3).Info("The workload needs admission checks updates", "clusterQueue", klog.KRef("", cqName), "admissionChecks", queueAdmissionChecks)
+		log.V(3).Info("The workload needs admission checks updates", "clusterQueue", klog.KRef("", cqName), "admissionChecks", admissionChecks)
 		wl.Status.AdmissionChecks = newChecks
 		err := r.client.Status().Update(ctx, wl)
 		return true, client.IgnoreNotFound(err)
 	}
 	return false, nil
 }
 
+// getChecksFromACStrategies return AdmissionChecks that are assigned to ResourceFlavors assigned to the Workload.
+func getChecksFromACStrategies(ctx context.Context, wl *kueue.Workload, queue *kueue.ClusterQueue) []string {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Kueue sets AdmissionChecks first based on ClusterQueue configuration and at this point Workload has no
+	// ResourceFlavors assigned, so we cannot match AdmissionChecks to ResourceFlavor.
+	// After QuotaReservation is, another reconciliation happens and we can match AdmissionChecks to ResourceFlavors
+	if wl.Status.Admission == nil {
+		log.V(2).Info("Workload has no Admission", "Workload", wl)
+		return []string{}
+	}
+
+	wlFlavors := make([]kueue.ResourceFlavorReference, 0)
```

Comment:
```suggestion
	wlFlavors := make([]kueue.ResourceFlavorReference)
```

Reply:
Could you clarify that please? It doesn't compile with that change, and the initial capacity of the wlFlavors is a bit tedious to calculate upfront